### ISSUE TITLE
Multi-select

### ISSUE THREAD

**Issue by charleslaw** [react-toolbox/react-toolbox#1880]
Thanks for this great library!  The components look great and it's been really easy to use.

There is 1 component that I cannot (easily) port over to using react-toolbox, and I would like to get it added to react toolbox, it's a multi select dropdown.  I am currently using https://github.com/Khan/react-multi-select which works well, but looks out of place with the rest of the UI.

Assuming it was built, is this something that could get added?  Should this be a separate component or something built using special options on Select?  I am not sure I could build it anytime soon, but if I do get a chance, I'd like to find out where it could fit in.

Thanks!

**Comment by hansena** [comment]
@charleslaw this is effectively the same functionality as the autocomplete...  is it just the display that you want to change?

**Comment by charleslaw** [comment]
Yes, I think that's a good way to describe it.  The display would be pretty different, and the behaviour would be different in some ways too.  Putting it into a dream list, I come up with:

* Selecting an item doesn't close the dropdown menu, so if selected/deselecting multiple items, the menu doesn't have to be re-opened
* There would be the addition of a "Select/Deselect All" option at the top of the dropdown
* The selected values display in 1 line - if all values are selected, the display shows "All items selected".  If a couple items are selected, it displays each item.  When the items fill 1 line, it switches from showing values individually to displaying "N items selected".
* The dropdown shows all the options and has some indication of which items are selected like a checkbox or using highlighting.  This is just personal preference I like how the filters in Google Sheets/Excel work where all the options are visible
* When not being used, the display fits in 1 line.  When closed, only the selected values or special message ("N items selected") displays.  The text input search box shows up as part of the dropdown list.

I think every options except the last one could easily fit with autocomplete and be configured via flags.  The last one could fit in there too, but I'm guessing would require a pretty big overhaul
